src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/orchestrator/CreateCaseTest.java

```diff
@@ -0,0 +1,177 @@
+        // given
+        CaseDetails exceptionRecord = createExceptionRecord("envelopes/exception-with-evidence.json");
+
+        // when
+        String caseCcdId = invokeCallbackEndpoint(exceptionRecord, "ccdCaseReference");
+
+        // then
+        await("Case is created")
+            .atMost(60, TimeUnit.SECONDS)
+            .pollDelay(2, TimeUnit.SECONDS)
+            .until(() -> isCaseCreated(caseCcdId, exceptionRecord));
+    }
+
+    /**
+     * Hits the services callback endpoint with a request to create case upon an exception record.
+     */
+    private String invokeCallbackEndpoint(
+        CaseDetails exceptionRecord,
+        String searchCaseReferenceType
+    ) throws IOException {
+        Map<String, Object> exceptionRecordDataWithSearchFields = exceptionRecordDataWithSearchFields(
+            exceptionRecord,
+            searchCaseReferenceType
+        );
+
+        CaseDetails exceptionRecordWithSearchFields =
+            exceptionRecord.toBuilder().data(exceptionRecordDataWithSearchFields).build();
+
+        CallbackRequest callbackRequest = CallbackRequest
+            .builder()
+            .eventId("createCase")
+            .caseDetails(exceptionRecordWithSearchFields)
+            .build();
+
+        CcdAuthenticator ccdAuthenticator = ccdAuthenticatorFactory.createForJurisdiction("BULKSCAN");
+
+        Response response = RestAssured
+            .given()
+            .relaxedHTTPSValidation()
+            .baseUri(testUrl)
+            .header(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+            .header(SyntheticHeaders.SYNTHETIC_TEST_SOURCE, "Bulk Scan Orchestrator Functional test")
+            .header(AUTHORIZATION, ccdAuthenticator.getUserToken())
+            .header(CcdCallbackController.USER_ID, ccdAuthenticator.getUserDetails().getId())
+            .body(callbackRequest)
+            .when()
+            .post("/callback/create-new-case")
+            .andReturn();
+
+        return getCaseCcdId(response);
+    }
+
+    private String getCaseCcdId(Response response) throws IOException {
+        assertThat(response.getStatusCode()).isEqualTo(200);
+
+        final AboutToStartOrSubmitCallbackResponse callbackResponse =
+            new ObjectMapper().readValue(response.getBody().asString(), AboutToStartOrSubmitCallbackResponse.class);
+        return (String)callbackResponse.getData().get("CASE_REFERENCE");
+    }
+
+    private Map<String, Object> exceptionRecordDataWithSearchFields(
+        CaseDetails exceptionRecord,
+        String searchCaseReferenceType
+    ) {
+        Map<String, Object> exceptionRecordData = new HashMap<>(exceptionRecord.getData());
+        exceptionRecordData.put("searchCaseReferenceType", searchCaseReferenceType);
+        return exceptionRecordData;
+    }
+
+    private CaseDetails createExceptionRecord(String resourceName) throws Exception {
+        UUID poBox = UUID.randomUUID();
```

Comment:
Since `envelopeId` is present in case data we should start using it. `poBox` randomisation was a hack back in a day. But! Best to do it outside of this PR